src/Microsoft.Identity.Web/AppServicesAuth/AppServicesAuthenticationHandler.cs

```diff
@@ -35,28 +35,15 @@ public class AppServicesAuthenticationHandler : AuthenticationHandler<AppService
         }
 
         // Constants
-        private const string AppServicesAuthIdTokenHeader = "X-MS-TOKEN-AAD-ID-TOKEN";
-        private const string AppServicesAuthIdpTokenHeader = "X-MS-CLIENT-PRINCIPAL-IDP";
 
         /// <inheritdoc/>
         protected override Task<AuthenticateResult> HandleAuthenticateAsync()
         {
             if (AppServicesAuthenticationInformation.IsAppServicesAadAuthenticationEnabled)
             {
-                string? idToken = GetIdToken();
-                string? idp = GetIdp();
-
-                if (idToken != null && idp != null)
+                ClaimsPrincipal claimsPrincipal = AppServicesAuthenticationInformation.GetUser(Context.Request.Headers);
```

Comment:
Most was moved to AppServicesAuthenticationInformation as it's also used in AppServicesAuthenticationTokenAcquisition now.